src/modules/job-list/stats.c

```diff
@@ -157,12 +210,69 @@ static json_t *stats_encode (struct job_stats *stats)
         return NULL;
     }
     json_decref (states);
+
+    if (name) {
+        json_t *no = json_string (name);
+        if (!no || json_object_set (o, "name", no) < 0) {
+            json_decref (no);
+            json_decref (o);
+            errno = ENOMEM;
+            return NULL;
+        }
+    }
     return o;
 }
 
+static int queue_stats_encode (struct job_stats_ctx *statsctx,
+                               json_t *o)
```

Comment:
The function name does not suggest that the stats are going to be inserted in `o`.  Maybe have the caller do that?